apps/api/src/types/index.ts

```diff
@@ -469,7 +469,9 @@ export type Agent = z.infer<typeof AgentSchema>;
 /**
  * Pulic Agent information Object, omits apiKey
  */
-export const AgentPublicSchema = AgentSchema.omit({ apiKey: true });
+export const AgentPublicSchema = AgentSchema.omit({ apiKey: true }).extend({
```

Comment:
We now have one type and one matching spec schema.  This is assuming we keep the spec up to date with the type.